pandas/_libs/tslibs/timestamps.pyi

```diff
@@ -124,7 +124,7 @@ class Timestamp(datetime):
         microsecond: int = ...,
         tzinfo: _tzinfo | None = ...,
         fold: int = ...,
-    ) -> datetime: ...
+    ) -> Timestamp: ...
```

Comment:
`pd.Timestamp.replace()` returns a `pd.Timestamp`. Mypy was also expecting it to return a `pd.Timestamp`.

Reply:
in typeshed, the parent class, `datetime` returns `Self` and alias for an unconstrained typevar.

Reply:
Can we already use Self with 3.8+__future__? If not, we could simply use a TypeVar bound by Timestamp and use it for self and the return value.

Reply:
> Can we already use Self with 3.8+**future**?

not sure. will need to check. in the past we have always needed to create bounded typevars to maintain type for class methods returning same type as self, but maybe this is no longer needed. will review our current usage at some point.

Reply:
Seems to come in python 3.11 https://peps.python.org/pep-0673/ not sure whether it will be/is backported with `__future__`

Reply:
used a TypeVar